javarosa/src/test/java/org/javarosa/xpath/test/XPathEvalTest.java

```diff
@@ -68,6 +68,8 @@ private void testEval(String expr, FormInstance model, EvaluationContext ec, Obj
                     fail("Doubles outside of tolerance [" + o + "," + t + " ]");
                 } else if (Double.isNaN(o) && !Double.isNaN(t)) {
                     fail("Result was NaN when not expected");
+                } else if (Double.isNaN(t) && !Double.isNaN(o)) {
```

Comment:
Good catch, not sure, I think we evaluate equality on these internally when we do xpath expressions, so I can't think of any.